# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1327110030 28800
#      Fri Jan 20 17:40:30 2012 -0800
# Branch stable
# Node ID 1facbe04b00c5510a0d9c861e8eb0449c9924ff7
# Parent  846273dae16ba2b04aa3f4191a6c6feecb92905c
Fix regression in 9fc597693b0b preventing updating of waitbar text string.

* waitbar.m: Fix regression in 9fc597693b0b preventing updating of
waitbar text string.

diff --git a/scripts/plot/waitbar.m b/scripts/plot/waitbar.m
--- a/scripts/plot/waitbar.m
+++ b/scripts/plot/waitbar.m
@@ -87,18 +87,22 @@ function retval = waitbar (varargin)
 
   if (h)
     p = findobj (h, "type", "patch");
     set (p, "xdata", [0; frac; frac; 0]);
     ax = findobj (h, "type", "axes");
     if (ischar (msg) || iscellstr (msg))
       th = get (ax, "title");
       curr_msg = get (th, "string");
+      ## graphics handles always store data as column vectors
+      if (iscellstr (msg))
+        msg = msg(:);  
+      endif
       cmp = strcmp (msg, curr_msg);
-      if (all (cmp(:)))
+      if (! all (cmp(:)))
         set (th, "string", msg);
       endif
     endif
   else
     h = __go_figure__ (NaN, "position", [250, 500, 400, 100],
                        "numbertitle", "off",
                        "toolbar", "none", "menubar", "none",
                        "integerhandle", "off",
